spec/controllers/queue_items_controller_spec.rb

```diff
@@ -100,4 +106,69 @@
     end
   end
   
+  describe "POST update_queue" do    
+    context "with valid input" do
+      let(:renee) {Fabricate(:user)}
+      let(:video) {Fabricate(:video)}
+      let(:item_alpha) {Fabricate(:queue_item, user: renee, position: 1, video: video)}
+      let(:item_beta) {Fabricate(:queue_item, user: renee, position: 2, video: video)}
+      before {session[:user_id] = renee.id}
+      
+      it "redirects to my queue page" do
+        post :update_queue, queue_items: [{id: item_alpha.id, position: 2}, {id: item_beta.id, position: 1}]
+        expect(response).to redirect_to my_queue_path
+      end
+      
+      it "reorders queue items" do
+        post :update_queue, queue_items: [{id: item_alpha.id, position: 2}, {id: item_beta.id, position: 1}]
+        expect(renee.queue_items).to eq([item_beta, item_alpha])
+      end
+      
+      it "normalizes the position numbers" do
+        post :update_queue, queue_items: [{id: item_alpha.id, position: 3}, {id: item_beta.id, position: 2}]
+        expect(renee.queue_items.map(&:position)).to eq([1, 2])
+      end
+    end
+    
+    context "with invalid input" do
+      let(:renee) {Fabricate(:user)}
+      let(:video) {Fabricate(:video)}
+      let(:item_alpha) {Fabricate(:queue_item, user: renee, position: 1, video: video)}
+      let(:item_beta) {Fabricate(:queue_item, user: renee, position: 2, video: video)}
+      before do 
+        session[:user_id] = renee.id
+        post :update_queue, queue_items: [{id: item_alpha.id, position: 3}, {id: item_beta.id, position: "B"}]
+      end
+      
+      it {is_expected.to redirect_to my_queue_path}      
+      it {is_expected.to set_flash['danger']}
+      
+      it "does not change queue items" do
+        expect(item_alpha.reload.position).to eq(1)
+        expect(item_beta.reload.position).to eq(2)
+      end
+    end
+    
+    context "with unauthenticated user" do
+      it "redirects to the front page" do
+        post :update_queue, queue_items: [{id: 1, position: 3}, {id: 2, position: 5}]
+        expect(response).to redirect_to root_path
+      end
+    end
+    
+    context "with queue items that do not belong to current user" do
+      it "does not change queue items" do
+        renee = Fabricate(:user)
+        bob = Fabricate(:user)
+        video = Fabricate(:video)
+        session[:user_id] = renee.id
+        item_alpha = Fabricate(:queue_item, user: bob, position: 1, video: video)
+        item_beta = Fabricate(:queue_item, user: renee, position: 2, video: video)
```

Comment:
alpha is created by user bob, beta by current user renee. renee attempts to update both her own item and bobs item. her item updates, his does not. i removed this because it seems that it is doing too much.